Scripts/CopyClassicVHDFromVMToLab.ps1

```diff
@@ -0,0 +1,253 @@
+        SourceUri = ''
+        StorageAccountKey = ''
+        StorageAccountName = ''
+    }
+
+    $disk = Get-AzureDisk | ? { $_.AttachedTo.RoleName -eq "$Name" }
+    $vmCopyContext.SourceUri = $disk.MediaLink.AbsoluteUri
+    $vmCopyContext.StorageAccountName = $disk.MediaLink.Host.Split('.')[0]
+    $vmCopyContext.StorageAccountKey = (Get-AzureStorageKey -StorageAccountName $vmStorageAccountName).Primary
+        
+    return $vmCopyContext
+}
+
+function Stop-AzureDtlVirtualMachine
+{
+    [CmdletBinding()]
+    param(
+        $VM
+    )
+
+    $classicVm = Get-AzureVM | Where-Object { $_.Name -eq $VM.Name }
+    Stop-AzureVM -Name $classicVm.Name -ServiceName $classicVm.ServiceName -Force | Out-Null
+}
+
+function Copy-AzureDtlVirtualMachineVhd
+{
+    [CmdletBinding()]
+    param(
+        $CopyContext
+    )
+
+    Write-Host "  preparing the destination context ... " -NoNewline
+    $destContext = New-AzureStorageContext -StorageAccountName $CopyContext.LabStorageAccountName -StorageAccountKey $CopyContext.LabStorageAccountKey
+    Write-Host $Done
+
+    $destContainerName = 'uploads'
+    Write-Host "  ensuring destination container '$destContainerName' exists ... " -NoNewline
+    New-AzureStorageContainer -Name $destContainerName -Context $destContext -Permission Off -ErrorAction SilentlyContinue | Out-Null
+    Write-Host $Done
+
+    if ($CopyContext.IsVMDiskManaged)
+    {
+        Write-Host "  copying managed disk ... " -NoNewline
+        $copyHandle = Start-AzureStorageBlobCopy -AbsoluteUri $CopyContext.VMSourceUri -DestContainer $destContainerName -DestBlob $CopyContext.VHDFileName -DestContext $destContext
+    }
+    else
+    {
+        Write-Host "  preparing the source context ... " -NoNewline
+        $srcContext = New-AzureStorageContext -StorageAccountName $CopyContext.VMStorageAccountName -StorageAccountKey $CopyContext.VMStorageAccountKey
+        Write-Host $Done
+
+        Write-Host "  copying unmanaged disk ... " -NoNewline
+        $copyHandle = Start-AzureStorageBlobCopy -AbsoluteUri $CopyContext.VMSourceUri -Context $srcContext -DestContainer $destContainerName -DestBlob $CopyContext.VHDFileName -DestContext $destContext
+    }
+
+    $copyStatus = $copyHandle | Get-AzureStorageBlobCopyState
+    while ($copyStatus.Status -eq "Pending")
+    {
+        $copyStatus = $copyHandle | Get-AzureStorageBlobCopyState 
+        $perComplete = ($copyStatus.BytesCopied / $copyStatus.TotalBytes) * 100
+        Write-Progress -Activity "Copying blob ... " -Status "Percentage Complete" -PercentComplete "$perComplete"
+        Start-Sleep 10
+    }
+
+    Write-Host $Done
+
+    return $copyStatus
+}
+
+###################################################################################################
+#
+# Main execution block.
+#
+
+try
+{
+    Write-Host "Selecting subscription '$SubscriptionId' ... " -NoNewline
+    Select-AzureSubscription -SubscriptionId $SubscriptionId -ErrorAction SilentlyContinue | Out-Null
```

Comment:
should we silently continue?

Reply:
We used to need it when we had both scripts in one. Now that they are separate, I can change it. In this case, it makes sense that both are correctly set.